File: lib/src/main/java/com/easy/lib/weight/banner/LoopBannerHolder.java
package com.easy.lib.weight.banner;


import android.view.View;

import java.util.Map;

/**
 * @author liu
 * @date 2017/11/8
 */
abstract public class LoopBannerHolder<T> {

    abstract public View createView();

    abstract public void bindData(View itemView, int position, T data);

    public View bindView(int position, T data) {
        View itemView = getView();
        bindData(itemView, position, data);
        return itemView;
    }

    private View getView() {

        Map<View, Boolean> viewMap = BannerHelper.getInstance().getBannerViewMap();

        for (View key : viewMap.keySet()) {
            if (viewMap.get(key) == false) {
                viewMap.put(key, true);
                return key;
            }
        }

        View view = createView();
        viewMap.put(view, true);
        return view;
    }


}

File: lib/src/main/java/com/easy/lib/weight/banner/LoopBannerHolder.kt
package com.easy.lib.weight.banner

import android.view.View

/**
 * @author liu
 * @date 2017/11/8
 */
abstract class LoopBannerHolder<T> {
    abstract fun createView(): View
    abstract fun bindData(itemView: View?, position: Int, data: T)
    fun bindView(position: Int, data: T): View {
        val itemView = view
        bindData(itemView, position, data)
        return itemView
    }

    private val view: View
        private get() {
            val viewMap =
                BannerHelper.bannerViewMap
            for (key in viewMap.keys) {
                if (viewMap[key] == false) {
                    viewMap[key] = true
                    return key
                }
            }
            val view = createView()
            viewMap[view] = true
            return view
        }
}